### DISCUSSION TITLE
BLE binding in bonding pair mode instead of pin

### DISCUSSION THREAD

**Discussion by jpbusman** [meshtastic/firmware#5329] [Ideas]
For connecting to a Garmin device as Node client (I am developing a ConnectIQ app) I need to pair without the pin logic.
Putting the device in pair mode and setup a secure connection (mode should be enabled after flashing, with button or serial by web interface)
I think this is the modern way for pairing.

**Comment by garthvh** [comment]
Use the just works mode and under Bluetooth config

**Reply by jpbusman** [reply]
Thanks, I get a pair succes.
Now I want to send to the characteristic but only 20 bytes can be send and received in one write or read (Garmin limited it)
I worked with dataparts to a Gopro before, can we implement this on a lora device?
For example 
packet 1: [total bytes to send] [bytecount] [bytes......]
packet 2: [bytecount] [bytes......]
packet 3: [bytecount] [bytes......]

**Reply by garthvh** [reply]
Meshtastic sends a lot more data than that, seems like a ton of work to break everything down to that small of pieces, a single packet would need ten chunks or more

**Reply by jpbusman** [reply]
Having a remote screen and cliient like a garmin device is a very nice option, please consider this option.

**Reply by jpbusman** [reply]
I have found another option is BLE is really not possible, my Garmin Fenix 7X has also Wifi, but to connect to the Lora, the Lora needs to act as an Accesspoint because out there in the wild is no AP and no internet.
So can the Wifi be extended with a checkbox 'Act as an Accesspoint' with DHCP module and set the IP range
Security None, WEB, WPA or WPA2

**Reply by fifieldt** [reply]
Unfortunately, AP management code is probably too big to fit in the flash of the devices we regularly see meshtastic on :(

**Reply by jpbusman** [reply]
Could you try, other option would be an esp32 accesspoint between garmin and Lora

**Reply by jpbusman** [reply]
What is the url for api/v1/fromradio for json output
And can i post the toradio options as json as a put?

**Reply by jpbusman** [reply]
I see more limitations in the GarminIQ SDK (I have limited memory and processor time)
I would be nice to solfe this on de Lora Side: 
only HTTP GET POST PUT and DELETE (no OPTIONS!)
Request Content types JSON or URL_ENCODED
Response Content types JSON, URL_ENCODED or plain text

**Reply by jpbusman** [reply]
> Meshtastic sends a lot more data than that, seems like a ton of work to break everything down to that small of pieces, a single packet would need ten chunks or more

Hello, I looked at the firmware code an suggest to add to new characteristics for the 20 byte communication
In the 2 new callbacks a loop for chuncks can be made, please have a look at it, it seems not to be that hard (I hope)
